ui/app/adapters/keymgmt/key.js

```diff
@@ -103,8 +103,13 @@ export default class KeymgmtKeyAdapter extends ApplicationAdapter {
   }
 
   async query(store, type, query) {
-    const { backend } = query;
-    return this.ajax(this.url(backend), 'GET', {
+    const { backend, provider } = query;
+    const providerAdapter = store.adapterFor('keymgmt/provider');
```

Comment:
I went back and forth on this. I need to populate the provider show route keys tab list with keys models. If I do this on the provider adapter I could do something like `store.push` and then `store.peekRecord` but that didn't seem ideal to me considering the query method on the keys adapter already returns keys models. Essentially we are querying keys by a provider.

Reply:
This isn't always true though, since the keys tab on the backend itself will have keys from any number of providers. This is an interesting point though, which is if the store is populated with *all* keys as they move through the UI, how do we filter it down for the keys tab on the provider view 🤔

Reply:
So on line 108 I do a check for `provider` in the query and then use a different url, `/:backend/:provider/key` which should only return keys for the given provider. Otherwise the `/:backend/key` endpoint is used to fetch all.